### PR TITLE
[Snyk] Security upgrade node from 14.16.1 to 16.20.2-bookworm-slim

### PR DESCRIPTION
**Pull request by yashasvibajpai**
<p>This PR was automatically created by Snyk using the credentials of a real user.</p><br />Keeping your Docker base image up-to-date means you’ll benefit from security fixes in the latest version of your chosen image.

#### Changes included in this PR 
Resolves INT-476

- Dockerfile

We recommend upgrading to `node:16.20.2-bookworm-slim`, as this image has only 29 known vulnerabilities. To do this, merge this pull request, then verify your application still works as expected.

Some of the most important vulnerabilities in your base image include:

| Severity                                                                                                                 | Priority Score / 1000  | Issue                                                                     | Exploit Maturity      |
| :------:                                                                                                                 | :--------------------  | :----                                                                     | :---------------      |
| critical severity   | **714**  | Out-of-bounds Read <br/>[SNYK-DEBIAN9-LIBWEBP-1289559](https://snyk.io/vuln/SNYK-DEBIAN9-LIBWEBP-1289559)   | No Known Exploit   |
| critical severity   | **714**  | Out-of-bounds Write <br/>[SNYK-DEBIAN9-LIBWEBP-1289560](https://snyk.io/vuln/SNYK-DEBIAN9-LIBWEBP-1289560)   | No Known Exploit   |
| critical severity   | **714**  | SQL Injection <br/>[SNYK-DEBIAN9-OPENLDAP-2808416](https://snyk.io/vuln/SNYK-DEBIAN9-OPENLDAP-2808416)   | No Known Exploit   |
| critical severity   | **714**  | SQL Injection <br/>[SNYK-DEBIAN9-OPENLDAP-2808416](https://snyk.io/vuln/SNYK-DEBIAN9-OPENLDAP-2808416)   | No Known Exploit   |
| critical severity   | **714**  | OS Command Injection <br/>[SNYK-DEBIAN9-OPENSSL-2807589](https://snyk.io/vuln/SNYK-DEBIAN9-OPENSSL-2807589)   | No Known Exploit   |

---

**Note:** _You are seeing this because you or someone else with access to this repository has authorized Snyk to open fix PRs._

For more information: 
🧐 [View latest project report](https://app.snyk.io/org/rudder-integrations/project/6935b8d1-633c-4b69-8cc2-4db53617dca1?utm_source&#x3D;github&amp;utm_medium&#x3D;referral&amp;page&#x3D;fix-pr)

🛠 [Adjust project settings](https://app.snyk.io/org/rudder-integrations/project/6935b8d1-633c-4b69-8cc2-4db53617dca1?utm_source&#x3D;github&amp;utm_medium&#x3D;referral&amp;page&#x3D;fix-pr/settings)

[//]: # 'snyk:metadata:{"prId":"c10363cb-1a5f-44f5-a900-e2db3480b417","prPublicId":"c10363cb-1a5f-44f5-a900-e2db3480b417","dependencies":[{"name":"node","from":"14.16.1","to":"16.20.2-bookworm-slim"}],"packageManager":"dockerfile","projectPublicId":"6935b8d1-633c-4b69-8cc2-4db53617dca1","projectUrl":"https://app.snyk.io/org/rudder-integrations/project/6935b8d1-633c-4b69-8cc2-4db53617dca1?utm_source=github&utm_medium=referral&page=fix-pr","type":"user-initiated","patch":[],"vulns":["SNYK-DEBIAN9-LIBWEBP-1289559","SNYK-DEBIAN9-LIBWEBP-1289560","SNYK-DEBIAN9-OPENLDAP-2808416","SNYK-DEBIAN9-OPENSSL-2807589"],"upgrade":["SNYK-DEBIAN9-LIBWEBP-1289559","SNYK-DEBIAN9-LIBWEBP-1289560","SNYK-DEBIAN9-OPENLDAP-2808416","SNYK-DEBIAN9-OPENLDAP-2808416","SNYK-DEBIAN9-OPENSSL-2807589"],"isBreakingChange":false,"env":"prod","prType":"fix","templateVariants":["updated-fix-title","priorityScore"],"priorityScoreList":[714,714,714,714],"remediationStrategy":"vuln"}'

---

**Learn how to fix vulnerabilities with free interactive lessons:**

 🦉 [SQL Injection](https://learn.snyk.io/lesson/sql-injection/?loc&#x3D;fix-pr)

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,4 +1,4 @@
-FROM node:14.16.1
+FROM node:16.20.2-bookworm-slim
```
Comment (krishna2020) [inline Dockerfile:1]: why this custom image `bookworm-slim` instead of plain 16.20.2 ?

Comment (krishna2020) [reply Dockerfile:1]: have we compared the vulnerabilities that are getting fixed in `16.20.2` vs `16.20.2-bookworm-slim` ?

Comment (krishna2020) [reply Dockerfile:1]: **node:<version>-slim**
This image does not contain the common packages contained in the default tag and only contains the minimal packages needed to run node. Unless you are working in an environment where only the node image will be deployed and you have space constraints, we highly recommend using the default image of this repository.
Ref: https://hub.docker.com/_/node/

Comment (krishna2020) [reply Dockerfile:1]: we can also explore updating to latest node 20.x version or at least 18.x ?

Comment (yashasvibajpai) [reply Dockerfile:1]: Thank you for pointing it out. This is an automated PR created by Snyk based on the specific version `node:16.20.2-bookworm-slim` .

**Comment by yashasvibajpai** [comment]
Closing this for the other upgrade PR.

**Pull request closed without merging**